Geospatial/OGR/SpatialWriter.cs

```diff
@@ -0,0 +1,132 @@
+﻿using OSGeo.OGR;
+using OSGeo.OSR;
+using USACE.HEC.Results;
+
+namespace Geospatial.OGR;
+
+
+public class SpatialWriter : IResultsWriter
+{
+  private Layer _layer;
+  private DataSource _dataSource;
+  private SpatialReference _srs;
+  private SpatialReference _dst;
+  private bool _headersWritten;
+  private string _xField;
+  private string _yField;
+
+  /// <summary>
+  /// SpatialWriter writes spatial results data. X and Y Fields are the ResultItem names of the x and y coordinates. 
+  /// </summary>
+  public SpatialWriter(string outputPath, string driverName, int projection, string xField, string yField)
+  {
+    _dataSource = Ogr.GetDriverByName(driverName).CreateDataSource(outputPath, null) ?? throw new Exception("Failed to create data source.");
+    _srs = new SpatialReference("");
+    _srs.ImportFromEPSG(4326); // WGS84
```

Comment:
What if our source isn't in WGS84?

Reply:
I think I just assumed it would be because the NSI is, but I can add a parameter for that